tests/karma.js

```diff
@@ -2151,6 +2155,1028 @@ beforeAll(function (done) {
+              '  <itemsperpage></itemsperpage>\n' +
+              ' </meta>\n' +
+              ' <data/>\n' +
+              '</ocs>'
+          }
+        })
+      }
+    }
+    )
+    .then(() => {
+      const shareids = [14, 18, 19]
+      var file = ['/test123.txt', '/test space and + and #123.txt', '/文件123.txt']
+      for (var i = 0; i < file.length; i++) {
+        provider.addInteraction({
+          uponReceiving: 'a GET request for an existent share',
+          withRequest: {
+            method: 'GET',
+            path: Pact.Matchers.term({
+              matcher: '.*\\/ocs\\/v1\\.php\\/apps\\/files_sharing\\/api\\/v1\\/shares\\/' + shareids[i] + '$',
+              generate: '/ocs/v1.php/apps/files_sharing/api/v1/shares/' + shareids[i]
+            }),
+            headers: validAuthHeaders
+          },
+          willRespondWith: {
+            status: 200,
+            headers: xmlResponseHeaders,
+            body: '<?xml version="1.0"?>\n' +
+              '<ocs>\n' +
+              ' <meta>\n' +
+              '  <status>ok</status>\n' +
+              '  <statuscode>100</statuscode>\n' +
+              '  <message/>\n' +
+              '  <totalitems></totalitems>\n' +
+              '  <itemsperpage></itemsperpage>\n' +
+              ' </meta>\n' +
+              ' <data>\n' +
+              '  <element>\n' +
+              '   <id>' + shareids[i] + '</id>\n' +
+              '   <share_type>0</share_type>\n' +
+              '   <uid_owner>admin</uid_owner>\n' +
+              '   <displayname_owner>admin</displayname_owner>\n' +
+              '   <permissions>19</permissions>\n' +
+              '   <parent/>\n' +
+              '   <expiration/>\n' +
+              '   <token/>\n' +
+              '   <uid_file_owner>admin</uid_file_owner>\n' +
+              '   <displayname_file_owner>admin</displayname_file_owner>\n' +
+              '   <additional_info_owner/>\n' +
+              '   <additional_info_file_owner/>\n' +
+              '   <path>' + file[i] + '</path>\n' +
+              '   <item_type>file</item_type>\n' +
+              '   <file_target>' + file[i] + '</file_target>\n' +
+              '   <share_with>test123</share_with>\n' +
+              '   <share_with_displayname>test123</share_with_displayname>\n' +
+              '   <share_with_additional_info/>\n' +
+              '   <attributes/>\n' +
+              '  </element>\n' +
+              ' </data>\n' +
+              '</ocs>'
+          }
+        })
+      }
+    }
+    )
+    .then(() => {
+      const files = ['test123.txt', '%E6%96%87%E4%BB%B6123.txt', 'test%20space%20and%20%2B%20and%20%23123.txt', 'newFileCreated123']
+      for (const file of files) {
+        provider.addInteraction({
+          uponReceiving: 'a DELETE request for a file',
+          withRequest: {
+            method: 'DELETE',
+            path: Pact.Matchers.term({
+              matcher: '.*\\/remote\\.php\\/webdav\\/' + file + '$',
```

Comment:
```suggestion
              matcher: '.*\\/remote\\.php\\/webdav\\/*',
```
Is it ok to use matcher like this so that files with any filename can be deleted!

Reply:
this gives error about multiple interactions. this can be fixed once the refactoring work comes in progress